fdbclient/include/fdbclient/MetaclusterManagement.actor.h

```diff
@@ -1581,9 +1581,14 @@ Future<std::vector<std::pair<TenantName, TenantMapEntry>>> listTenants(
 		try {
 			tr->setOption(FDBTransactionOptions::READ_SYSTEM_KEYS);
 			tr->setOption(FDBTransactionOptions::READ_LOCK_AWARE);
-			if (offset == 0 && filters.empty()) {
-				std::vector<std::pair<TenantName, TenantMapEntry>> tenants =
-				    wait(listTenantsTransaction(tr, begin, end, limit));
+			if (filters.empty()) {
```

Comment:
I know this was my suggestion, but now that I'm thinking about it again I realize that there is a potential risk if somebody uses a big offset that they may not be able to actually list everything. To be honest, I think if you are doing something like that then it would be a misuse and you should really specify different query bounds. Perhaps this is alright then.